src/CqlProcessor.js

```diff
@@ -50,20 +53,29 @@ export default class CqlProcessor {
    * Evaluate an expression from the CQL library represented by elmJson against
    * the patient bundle.
    * @param {string} expr - The name of an expression from elmJson
+   * @param {string} executionDateTime - Optional, ISO formatted execution datetime if not now
    * @returns {object} results - The results from executing the expression
    */
-  async evaluateExpression(expr) {
+  async evaluateExpression(expr, executionDateTime = undefined) {
     // Only try to evaluate an expression if we have a patient bundle loaded.
     if (this.patientSource._bundles && this.patientSource._bundles.length > 0) {
       let results;
       if (expr == "__evaluate_library__") {
-        results = await this.executor.exec(this.patientSource);
+        results = executionDateTime != undefined ? 
+          await this.executor.exec(this.patientSource, cql.DateTime.parse(executionDateTime)):
+          await this.executor.exec(this.patientSource);
         return results.patientResults[this.patientID];
-      } else {
-        results = await this.executor.exec_expression(
-          expr,
-          this.patientSource
-        );
+      } else {     
+        results = executionDateTime != undefined ? 
+          await this.executor.exec_expression(
+            expr,
+            this.patientSource,
+            cql.DateTime.parse(executionDateTime)
+          ):
+          await this.executor.exec_expression(
+            expr,
+            this.patientSource
+          );
```

Comment:
Not that it matters too much but is there a need to check if executionDateTime is undefined?  If it is not passed in the executor just takes the undefined argument and adds it to the PatientContext. I other words, I think passing in an undefined value results in the same outcome as checking it and not passing it in.

Reply:
I think the check is needing for parsing - the executor takes a cql.DateTime object, and executionDateTime is a string. I don't think the parse function handles undefined gracefully.